src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,8 +1,52 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final byte OPERATION_TYPE_INDEX = 0;
+    private static final byte AMMOUNT_INDEX = 1;
+    private int sumSupply = 0;
+    private int sumBuy = 0;
 
     public void getStatistic(String fromFileName, String toFileName) {
+        readFromFile(fromFileName);
+        writeToFile(toFileName);
+    }
+
+    private void readFromFile(String fromFileName) {
+        File file = new File(fromFileName);
+        try (BufferedReader reader = new BufferedReader(new FileReader(file))) {
+            String value = reader.readLine();
+            while (value != null) {
+                String[] data = value.split(",");
+                if (data[OPERATION_TYPE_INDEX].equals("buy")) {
+                    sumBuy += Integer.parseInt(data[AMMOUNT_INDEX]);
+                } else if (data[OPERATION_TYPE_INDEX].equals("supply")) {
+                    sumSupply += Integer.parseInt(data[AMMOUNT_INDEX]);
+                }
+                value = reader.readLine();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can`t read file", e);
+        }
+    }
```

Comment:
you should split this method into two smaller ones: one for reading from the file that will be returning read data and the second one that will calculate sumBuy, sumSupply and after that call another separate method that will be forming a report that you should pass to the writeToFile() method